result/typetests.py

```diff
@@ -3,18 +3,15 @@
 from .result import Result, Ok, Err
 
 
-res1 = Result.Ok('hello')  # type: Result[str, int]
-if res1.is_ok():
+res1 = Ok('hello')  # type: Result[str, int]
+if isinstance(res1, Ok):
     ok = res1.ok()  # type: Optional[str]
```

Comment:
We can narrow our check here and assert the type has narrowed down to `Ok`,
```suggestion
    ok = res1  # type: Ok[str]
    okValue = res1.ok()  # type: str
```

Reply:
Fixed in 2ef68c749532376dc0124233a801a9f0f2224885